apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -1017,13 +1074,25 @@ internal BigtableClientImpl(BigtableServiceApiClient client, string appProfileId
             _appProfileId = appProfileId;
         }
 
+        // TODO: Add a public constructor after multi-channel support?
+        internal BigtableClientImpl(string appProfileId, params BigtableServiceApiClient[] clients)
+        {
+            _clients = clients;
+            _appProfileId = appProfileId;
+        }
+
         // TODO: Add Multi-channel support
         /// <inheritdoc/>
-        private BigtableServiceApiClient GetUnderlyingClient() => _client;
+        private BigtableServiceApiClient GetUnderlyingClient() => _clients.Length > 1 ? _clients[GetClientIndex()] : _clients[0];
```

Comment:
I would keep things simple here and always return `_clients[GetClientIndex()]`. I don't think the atomic update is going to have a significant impact on the overall operation when this result is then used to make an API call.